edxval/views.py

```diff
@@ -92,15 +95,69 @@ class VideoDetail(generics.RetrieveUpdateDestroyAPIView):
     serializer_class = VideoSerializer
 
 
-class SubtitleDetail(MultipleFieldLookupMixin, generics.RetrieveUpdateDestroyAPIView):
+class VideoTranscriptView(APIView):
     """
-    Gets a subtitle instance given its id
+    A Transcription View, used by VEDA to create video transcripts.
     """
     authentication_classes = (OAuth2Authentication, SessionAuthentication)
-    permission_classes = (ReadRestrictedDjangoModelPermissions,)
-    lookup_fields = ("video__edx_video_id", "language")
-    queryset = Subtitle.objects.all()
-    serializer_class = SubtitleSerializer
+
+    # noinspection PyMethodMayBeStatic
+    def post(self, request):
+        """
+        Creates a video transcript instance with the given information.
+
+        Arguments:
+            request: A WSGI request.
+        """
+        attrs = ('video_id', 'name', 'language_code', 'provider', 'file_format')
+        missing = [attr for attr in attrs if attr not in request.data]
+        if missing:
+            LOGGER.warn(
+                '[VAL] Required transcript params are missing. Request params are %s', request.data.keys()
+            )
```

Comment:
`Request params are %s', request.data.keys()` I think mentioning the missing param would be more helpful rather than specifying request data.